crates/frontend/src/api.rs

```diff
@@ -129,37 +107,25 @@ pub async fn delete_context(
     }
 }
 
-// #[server(GetExperiments, "/fxn", "GetJson")]
 pub async fn fetch_experiments(
-    exp_filters: ExpListFilters,
-    pagination_filters: ListFilters,
+    filters: &ExperimentListFilters,
+    pagination: &ListFilters,
     tenant: String,
 ) -> Result<PaginatedResponse<ExperimentResponse>, ServerFnError> {
     let client = reqwest::Client::new();
     let host = use_host_server();
+    let pagination = pagination.to_string();
 
-    let mut query_params = vec![];
-    if let Some(status) = exp_filters.status {
-        let status: Vec<String> = status.iter().map(|val| val.to_string()).collect();
-        query_params.push(format!("status={}", status.join(",")));
-    }
-    if let Some(from_date) = exp_filters.from_date {
-        query_params.push(format!("from_date={}", from_date));
-    }
-    if let Some(to_date) = exp_filters.to_date {
-        query_params.push(format!("to_date={}", to_date));
-    }
-    if let Some(page) = pagination_filters.page {
-        query_params.push(format!("page={}", page));
-    }
-    if let Some(count) = pagination_filters.count {
-        query_params.push(format!("count={}", count));
-    }
-    if let Some(all) = pagination_filters.all {
-        query_params.push(format!("all={}", all));
-    }
-
-    let url = format!("{}/experiments?{}", host, query_params.join("&"));
+    let url = if pagination.is_empty() {
+        format!("{}/experiments?{}", host, filters.to_string())
+    } else {
+        format!(
+            "{}/experiments?{}&{}",
+            host,
+            filters.to_string(),
+            pagination
+        )
```

Comment:
this does not seem correct, if someone does not send `filters`,  the url will become: `base_url/experimnets?&exp_filt1=abc`

Reply:
There are always date filters

Reply:
that is not being guaranteed by the type
all fields in `ExperimentListFilters` are optional, so one can simply avoid sending any params and then it breaks over here

Reply:
It is guaranteed by code, 
```
from_date: Utc.timestamp_opt(0, 0).single(),
        to_date: Utc.timestamp_opt(4130561031, 0).single(),
```
https://github.com/juspay/superposition/pull/280/files/64469a660bb91eabe003b378e3a0ec64d2de3961#diff-a26bd230250ef10be78fd6369216b71e556f5b808a9cd12e2be3b6551c6e8cbbR41

Doesn't need to be guaranteed by the type